### ISSUE TITLE
I have general ask regarding capturing conversion history and JSON output, so that we can avoid API call on follow up question

### ISSUE THREAD

**Issue by ajmal-yazdani** [jordanbean-msft/semantic-kernel-sleeping-bag#10]
Thanks for your work here. I learned a lot and here is my use case and question based on your work.

Question: I have a use case where I am using stepwise planner to call some external API's (bunch of API's calls in between, different steps) and the final output is a JSON something like (for example) {"name":"John", "Age": "18", "occupation": "Engineer"}. My first ask is "What is the age of John? and stepwise planner runs and api's calls done and and above JSON output comes as observation and I am getting answer in natural language. Now my concern is when I am asking follow up question like "what's is the occupation?"  then LLM saying don't know a and asking for more context and when I provide information/context then stepwise planner runs and again API call and finally getting answer. Can we use memory here and avoid API call?

**Comment by jordanbean-msft** [comment]
Yes, absolutely, my demo app was to demonstrate the power of the Stepwise planner. I did not put in optimizations or conversation history to keep the demo app simple. In the demo app, each conversation is independent. In a real-world app, you should store the conversation in memory (which is both faster and saves tokens).

**Comment by ajmal-yazdani** [comment]
> Yes, absolutely, my demo app was to demonstrate the power of the Stepwise planner. I did not put in optimizations or conversation history to keep the demo app simple. In the demo app, each conversation is independent. In a real-world app, you should store the conversation in memory (which is both faster and saves tokens).

Ok. Any good code article or blog you can refer please?
